src/Thruway/Role/Subscriber.php

```diff
@@ -219,13 +251,49 @@ public function subscribe(ClientSession $session, $topicName, callable $callback
             "options"    => $options,
             "deferred"   => $deferred
         ];
-
+        
         array_push($this->subscriptions, $subscription);
-
+        
         $subscribeMsg = new SubscribeMessage($requestId, $options, $topicName);
         $session->sendMessage($subscribeMsg);
 
         return $deferred->promise();
     }
-
+	
+    /**
+     * process unsubscribe
+     * @param ClientSession $session
+     * @param type $topicName
+     * @return boolean
+     */
+    public function unsubscribe(ClientSession $session, $topicName)
+    {
+        $requestId = Utils::getUniqueId();
+        $subscriptionId = false;
+        $deferred  = new Deferred();
+        
+        foreach ($this->subscriptions as $i => $subscription) {
+            if ($subscription["topic_name"] == $topicName) {
+                $subscriptionId = $subscription["subscription_id"];
+                
+                $this->subscriptions[$i]["unsubscribed_request_id"] = $requestId;
+                $this->subscriptions[$i]["unsubscribed_deferred"] = $deferred;
+            }
+        }
```

Comment:
While it would be good to keep the pattern that Autobahn|JS uses (which would have `subscribe` return a promise that resolves to a `Subscription` object), This would be making a change to the API. Right now the promise resolves with the `SubscribedMessage`.

If `unsubscribe` were implemented to take the `subscription_id`, the `SubscribedMessage` would have enough information.